### ISSUE TITLE
feat(train): support interval metric aggregation for disp_avg

### ISSUE THREAD

**Issue by OutisLi** [deepmodeling/deepmd-kit#5819]
## Summary

Implement `training.disp_avg` in the backend-independent training loop and enable it for pt_expt.

pt_expt currently accepts the option but reports only the latest step metrics. The legacy PT trainer has separate interval-aggregation logic, which should be replaced by one shared implementation.

## Root cause

The common trainer loop has no first-class metric aggregation policy. Consequently, each backend must either implement averaging independently or silently ignore `disp_avg`.

## Proposed design

Add a common `MetricAccumulator` abstraction that:

- accumulates detached sums and counts per task and metric;
- distinguishes training step averages from atom-weighted validation averages;
- supports sparse multi-task sampling without dividing by zero;
- resets only after a display interval is emitted;
- optionally reduces sums and counts across ranks at display time.

Backend-neutral aggregation should live under `deepmd/dpmodel/train`. Torch-specific tensor/reduction helpers may live under `deepmd/pt_expt/train`, with legacy PT importing the same implementation.

## Performance and correctness requirements

- Keep accumulated values on device where possible.
- Perform host synchronization only at display boundaries.
- Do not fetch extra training batches merely to display inactive tasks.
- Do not introduce operations into the compiled model graph.
- Preserve existing behavior when `disp_avg=false`.
- Preserve the unsampled-task safeguards covered by #5809.

## Acceptance criteria

- `disp_avg=true` reports the mean over the completed display interval.
- `disp_avg=false` reports the existing instantaneous metrics.
- Single-task and sparsely sampled multi-task cases are covered.
- No NaN or division by zero occurs when a task has zero samples in an interval.
- The accumulator resets correctly after regular and final partial intervals.
- Distributed aggregation uses global sums and counts with collectives only at display time.

Refs #5755.

**Comment by SchrodingersCattt** [comment]
One additional acceptance case may be needed for optional or partially labeled properties.

In a single-task PT training run with disp_avg=true, some systems contained E/F labels but no virial labels. DeePMD correctly set find_virial=0 and excluded the virial term from the optimized loss, while display_if_exist() returned NaN for the per-step mae_v.

The current interval accumulator adds this NaN directly:

```
train_loss_accu[item] += more_loss[item]
```

As a result, one batch without virial labels makes the displayed virial metric `NaN` for the entire interval, although all labeled virials, the total loss, model parameters, and validation metrics remain finite.

It may therefore be useful for `MetricAccumulator` to maintain a valid count per metric, excluding unavailable-property values according to `find_*` or an equivalent validity signal:

$$
\bar m_v =
\frac{\sum_i \mathbf{1}(\mathrm{find\_virial}_i)m_{v,i}}
{\sum_i \mathbf{1}(\mathrm{find\_virial}_i)}
$$

Only an interval with zero valid samples for that metric should report `NaN`.

We reproduced this with DeePMD-kit 3.2.0: legacy PT with `disp_avg=true` reported `mae_v=NaN` in nearly every interval, while PT-expt on the same mixed-label dataset only reported `NaN` when the displayed step itself sampled an E/F-only system. This seems directly relevant to the proposed per-metric sum/count abstraction.
